### ISSUE TITLE
Many broken links.

### ISSUE THREAD

**Issue by leamas** [opencpn-manuals/plugins#6]
Linkchecker reveals no less than 96 broken links in the plugins manual.  I ran `linkchecker -r8 docs/index.html`after a local build.

Attaching the logfile:  [linkcheck.log.gz](https://github.com/opencpn-manuals/plugins/files/6967571/linkcheck.log.gz)

**Comment by leamas** [comment]
I have added a "check links" step to the build. Example of output [here](https://github.com/opencpn-manuals/plugins/runs/3300514368?check_suite_focus=true). Note that the complete log is available as a link. Latest log is at  http://ix.io/3vEa

**Comment by Rasbats** [comment]
Errrr ... thanks. Guess my work on the manuals was not checked properly. Will need some work but easily checked with VS Code.
(Links in the pages)

**Comment by leamas** [comment]
There is also a w3c linkchecker. It's very slow, too slow for regular builds but finds more errors. I might try to find some way to use it optionally. Perhaps.

**Comment by Rasbats** [comment]
Each manual usually has a single page. This should be (relatively) easy to check.

**Comment by rgleason** [comment]
If these links are for screenshots, files and other content, we probably still have them in the user manual, and we can dig them out.

**Comment by Rasbats** [comment]
Yes ... I have a complete set of the files that are needed.

**Comment by Rasbats** [comment]
@leamas : Which linkchecker are you using locally?

**Comment by leamas** [comment]
Two. The first  is  'linkchecker' from  https://linkcheck.github.io/linkchecker/. This is a quick test which is done on all builds. You can see the  results in the logs at for example https://github.com/opencpn-manuals/development/actions (i. e.,  under the Actions button).

The second is checklink from https://metacpan.org/release/W3C-LinkChecker. This is much slower, but finds more problems. Currently experimenting on the development manual. Here, you can add `[full-linkcheck]` to the commit message. The build will then run this; an example on results [here](https://github.com/opencpn-manuals/development/runs/3306111781?check_suite_focus=true).

**Comment by leamas** [comment]
Ah... local builds. I use the same there, both of them.

**Comment by leamas** [comment]
I have added new checks using the W3C-LinkChecker script. This is much more aggressive, and finds more errors. Some notes in INSTALL.md.

tl;dr Any commit with the string [full-linkcheck] in the commit message will run the new checks which takes about 20 minutes, See results under the [Actions button](https://github.com/opencpn-manuals/plugins/actions),

**Comment by Rasbats** [comment]
[full-linkcheckj]  should be  [full-linkcheck]  I think.

**Comment by leamas** [comment]
indeed "blushes"

**Comment by Rasbats** [comment]
Too many of my commits in the main branch.

Reread WORKFLOWS.adoc and now using testing branch.

**Comment by Rasbats** [comment]
@rgleason : Working on my many link errors. Would appreciate some help with checking the plugins where you host the published managed/non-managed versions (about 20). These are listed in master-plugins.txt and beta-plugins.txt.

My PR for chartscale_pi show the typical errors. Apart from missing images the rest are due to the use of link: instead of xref: for an intra document reference. VS Code does not allow a preview for these 'links', as it expects to find the link on the Internet website. However Firefox does show the xref link and is good for checking.

At this end I am starting with my published plugins.

Once we have done these intend updating sources.state and we can do another build of the manuals.

**Comment by leamas** [comment]
I have found a quick way to check for unresolved Antora links after making a local build, like this (assuming the html files lives in docs/):

    $ grep -r  'class="page unresolved"' docs

It's fast and  produces output like:

```
docs/AlternativeWorkflow/index.html:<p><a href="#Plugin-Dependencies.adoc" class="page unresolved">A Note on Plugin Dependencies</a></p>
docs/ocpn-dev-manual/5.3.1/pm-tp-language-nuance.html:<p><a href="#pi_installer_dev_setup.adoc" class="page unresolved">PI Manager Dev Setup</a>  FIXME: broken links</p>

```

The html filename is the same as the .adoc file besides the extension. I mostly use `git grep`  to dig further. The links outputted are  red if browsing the site with the browser.

**Comment by leamas** [comment]
None of the bad lilnks are visible when building the development manual. Thanks!

I guess the other errors should be fixed as well, but there is now no need from the development part i. e., it is possible to lower the priority.

**Comment by Rasbats** [comment]
A lot of bad links are due to "Back to top of page". The Contents sidebar takes care of this. Intend removing these intra document links unless anyone can see a reason for keeping them.

**Comment by leamas** [comment]
I recognize those, memories form when I made the first attempts to move from dokuwiki->Antora. I just removed them.

**Comment by leamas** [comment]
Now, something terrible has happened. Checking links in development manual (which rightfully now include plugins) I get the following:

```
Processing	file:///home/runner/work/development/development/docs/opencpn-plugins/0.1/index.html

This may take some time if the document has many links to check.


List of broken links and other issues:

file:///home/runner/work/development/development/docs/opencpn-plugins/0.1/index.html	
 Lines: 30, 32, 62, 67, 96, 99, 102, 111, 114, 117, 120, 123, 126, 129, 132, 135, 138, 141, 150, 153, 156, 159, 162, 171, 174, 177, 180, 183, 186, 189, 192, 201, 204, 207, 216, 219, 222, 225, 234, 237, 240, 243, 252, 255, 258, 261, 264, 267, 270, 273, 329, 361, 364, 395, 396
  Code: 200 (no message)
 To do: Some of the links to this resource point to broken URI fragments
	(such as index.html#fragment).
The following fragments need to be fixed:
	weather_routing:ROOT:weather_routing.adoc	Line: 243
	fugawi:ROOT:fugawi.adoc       	Line: 123
	odraw:ROOT:odraw.adoc         	Line: 201
	opencpn-master-plugins:twocan:twocan.adoc	Line: 270
	statusbar:ROOT:statusbar.adoc 	Line: 252
	radar:ROOT:Home.adoc          	Line: 99
	watchdog:ROOT:watchdog.adoc   	Line: 204
	oesenc:ROOT:oesenc.adoc       	Line: 120
	opencpn-master-plugins:squiddio:squiddio.adoc	Line: 171
	otcurrent:ROOT:otcurrent.adoc 	Line: 186
	tactics:ROOT:tactics.adoc     	Line: 216
	sar:ROOT:sar.adoc             	Line: 207
	vdr:ROOT:vdr.adoc             	Line: 159
	opencpn-master-plugins:wmm:wmm.adoc	Line: 192
	debugger:ROOT:debugger.adoc   	Line: 261
	googleearth:ROOT:googleearth.adoc	Line: 174
	s63_vector_charts:ROOT:s63_vector_charts.adoc	Line: 114
	find-it:ROOT:find-it.adoc     	Line: 156
	odometer:ROOT:odometer.adoc   	Line: 273
	rotationctrl:ROOT:rotationctrl.adoc	Line: 132
	nv_charts:ROOT:nv_charts.adoc 	Line: 111
	dead_reckoning:ROOT:dead_reckoning.adoc	Line: 183
	projections:ROOT:projections.adoc	Line: 141
	bsb4_charts:ROOT:bsb4_charts.adoc	Line: 117
	nmea_converter:ROOT:nmea_converter.adoc	Line: 162
	route_great_circle:ROOT:route_great_circle.adoc	Line: 180
	celestial_navigation:ROOT:celestial_navigation.adoc	Line: 177
	windvane:ROOT:windvane.adoc   	Line: 225
	dash-t:ROOT:dash-t.adoc       	Line: 150
	rtlsdr:ROOT:rtlsdr.adoc       	Line: 102
	pypilot:ROOT:pypilot.adoc     	Line: 264
	weatherfax:ROOT:weatherfax.adoc	Line: 234
	launcher:ROOT:launcher.adoc   	Line: 258
	calculator:ROOT:calculator.adoc	Line: 255
	vfkaps:ROOT:vfkaps.adoc       	Line: 129
	shipdriver:ROOT:shipdriver.adoc	Line: 267
	climatology:ROOT:climatology.adoc	Line: 240
	iacfleet:ROOT:iacfleet.adoc   	Line: 237
	polar:ROOT:polar.adoc         	Line: 222
	photolayer:ROOT:photolayer.adoc	Line: 126
	otidalplan:ROOT:otidalplan.adoc	Line: 329
	chartscale:ROOT:chartscale.adoc	Line: 135
	objsearch:ROOT:objsearch.adoc 	Line: 138
	ais_radar_display:ROOT:ais_radar_display.adoc	Line: 96
	opencpn-master-plugins:ge2kap:ge2kap.adoc	Line: 189
	logbook:ROOT:logbook.adoc     	Line: 153
	sweep_plot:ROOT:sweep_plot.adoc	Line: 219
Anchors

Found 2 anchors.
```

This is not as it used to be (?) What's happening?

**Comment by Rasbats** [comment]
@Rasbats : Is back.  Plugins manual after the last authoring commit looks sane. What commit is being used for the main manual?

>./plugins https://github.com/opencpn-manuals/plugins 4fe583e771fb8f92520f678871bbed59cfe2eba7

This commit:

```
Microsoft Windows [Version 6.3.9600]
(c) 2013 Microsoft Corporation. All rights reserved.

C:\Users\Mike>cd C:\Projects\opencpn-manuals\plugins

C:\Projects\opencpn-manuals\plugins>git show 4fe583e771fb8f92520f678871bbed59cfe2eba7
commit 4fe583e771fb8f92520f678871bbed59cfe2eba7
Author: Rasbats <miker@members.swis.net>
Date:   Mon Aug 9 10:47:09 2021 +0100

    Renaming

    Rename plugins/plugins to plugins/manual. At the same time change manual and manual-beta to master-plugins and beta-plugins

diff --git a/.gitignore b/.gitignore
index 8b13789..d8f8d46 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1 @@
-
+docs
diff --git a/docs/.nojekyll b/docs/.nojekyll
deleted file mode 100644
index e69de29..0000000
diff --git a/docs/404.html b/docs/404.html
deleted file mode 100644
index 7ccab17..0000000
--- a/docs/404.html
+++ /dev/null
:

commit 4fe583e771fb8f92520f678871bbed59cfe2eba7
Author: Rasbats <miker@members.swis.net>
Date:   Mon Aug 9 10:47:09 2021 +0100

    Renaming

    Rename plugins/plugins to plugins/manual. At the same time change manual and manual-beta to master-plugins and beta-plugins

diff --git a/.gitignore b/.gitignore
index 8b13789..d8f8d46 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1 @@
-
+docs
diff --git a/docs/.nojekyll b/docs/.nojekyll
deleted file mode 100644
index e69de29..0000000
diff --git a/docs/404.html b/docs/404.html
deleted file mode 100644
index 7ccab17..0000000
--- a/docs/404.html
+++ /dev/null
@@ -1,520 +0,0 @@
:
```

Is this name change the problem?

**Comment by leamas** [comment]
The name change was a problem, but it was fixed long ago. I'll try to make local tests to see if it's about the plugin manual or how it's referenced in development. Stay tuned.

**Comment by Rasbats** [comment]
How about the indentation in the start paths for the plugins directories?

**Comment by leamas** [comment]
I have done a local check on the output from plugins/site.yml, attached. It doesn't look good (:

[checklink.log.gz](https://github.com/opencpn-manuals/plugins/files/7002937/checklink.log.gz)

**Comment by leamas** [comment]
need some sleep...

**Comment by leamas** [comment]
Hm... looking at this excerpt from log file:
```
List of broken links and other issues:

file:///home/mk/src/opencpn-manuals/plugins/docs/opencpn-plugins/0.1/authoring/TODO.html	
 Lines: 7, 31, 33, 171, 189, 192, 270
  Code: 200 (no message)
 To do: Some of the links to this resource point to broken URI fragments
	(such as index.html#fragment).
The following fragments need to be fixed:
	opencpn-master-plugins:twocan:twocan.adoc	Line: 270
	opencpn-master-plugins:squiddio:squiddio.adoc	Line: 171
	opencpn-master-plugins:ge2kap:ge2kap.adoc	Line: 189
	opencpn-master-plugins:wmm:wmm.adoc	Line: 192
```

The other modules are referenced as for example _opencpn-master-plugins:twocan:twocan.adoc_ . However, when I look into  _manual/master-plugin/antora.yml_ I see
``` 
name: opencpn-plugins
title: OpenCPN Plugins
version: '0.1'
nav: 
```
That is, the reference above should be changed like `opencpn-master-plugins:twocan:twocan.adoc -> opencpn-plugins:twocan:twocan.adoc`

There are other options, I guess. One could update antora.yml to match opencpn-master-plugins. But I wonder if not the best might be to use the default value here i. e.,  `opencpn-master-plugins:twocan:twocan.adoc -> twocan:twocan.adoc`. This way it would work without changes when used in the beta-plugins context.

Thoughts?

**Comment by leamas** [comment]
Another problem seems to be about missing files. For example, there are references to launcher:launcher.adoc:

```
 git grep launcher.adoc
dokuwiki plugins.txt:xref:launcher:launcher.adoc[launcher] +
master-plugins/modules/ROOT/pages/plugins.adoc:* xref:launcher:ROOT:launcher.adoc[launcher #]  
master-plugins/modules/ROOT/partials/nav-plugins.adoc:** xref:launcher:ROOT:launcher.adoc[launcher]
```
But I cannot find any module launcher and no file launcher.adoc:

```
$ find . -name launcher.adoc
$
```

Have you some files locally that are not checked in to git?

**Comment by leamas** [comment]
... or have I missed to update the sources? Stay tuned..

**Comment by Rasbats** [comment]
site.yml has a bad page-resource url entry. This is resulting in many of the broken links. Fixing (I hope).

**Comment by leamas** [comment]
I have pushed a simple fix to nav-plugins.adoc, it fixes just a few errors. You need to rebase.

**Comment by oplaydo** [comment]
Have you removed the settings option for this repo? It was used as an easy route to the website!!!

**Comment by Rasbats** [comment]
Thanks for picking up the mistakes in nav-plugins.adoc. Now checking the log for the site.yml fix.

**Comment by Rasbats** [comment]
Settings is back.

**Comment by leamas** [comment]
> Have you removed the settings option for this repo? It was used as an easy route to the website!!!

Not that I'm aware of.... or did I?

> Settings is back.

OK. Time for a new build and check?

**Comment by leamas** [comment]
If you are talking about the settings at the very  bottom of site.yml they are only used to populate the ui links under Resources (unless you have been using them in the adoc files)

**Comment by Rasbats** [comment]
Red herring. Now checking a new build.

**Comment by leamas** [comment]
I pushed a typo fix (possibly left-over). Nothing important.

**Comment by Rasbats** [comment]
Thanks. Where in the build process does sources.state get used? Intending a source-state.sh update to check some of the dev manuals PR that have been accepted.

**Comment by Rasbats** [comment]
Here?

```
# Update dependencies and build site
if [ -f source-state.sh ]; then ./source-state.sh restore; fi
antora $PLAYBOOK
touch $SITE_DIR/.nojekyll
```

**Comment by leamas** [comment]
Yup

**Comment by Rasbats** [comment]
So if I just do source-state.sh update everything will be taken care of. Is that correct?

**Comment by leamas** [comment]
I wonder if the best way could be to remove all plugins, check that everything is fine and the add them in small batches, fixing errors before adding more?

**Comment by leamas** [comment]
> So if I just do source-state.sh update everything will be taken care of. Is that correct?

Basically, you need to check in sources.state.

**Comment by Rasbats** [comment]
>I wonder if the best way could be to remove all plugins, check that everything is fine and the add them in small batches, fixing errors before adding more?

The problem is that the PR for devs sometimes takes a long time to action. I have a list of all the plugins and I can test before the PR. 

I have a plugins manual that works, although it has broken links. Notice in the developer manual that when we go to plugins manual from development the home page (house symbol) always links to the plugins manual.

Is this a version issue? I have been using v0.1. Develoment uses v5.3.1.

**Comment by leamas** [comment]
> I have a plugins manual that works, although it has broken links.

Yes, and those links are visible also in  the developer manual. But yes, also from the developer manual things "works", sort of.

**Comment by leamas** [comment]
> Is this a version issue? I have been using v0.1. Develoment uses v5.3.1.

Not a problem, the overall linking is fine: https://opencpn-manuals.github.io/development

**Comment by leamas** [comment]
> The problem is that the PR for devs sometimes takes a long time to action. I have a list of all the plugins and I can test before the PR.

Yes, I can understand that. But if we want to came to something stable we need to fix the broken links in "your " repos, and then create PRs from those.

Question is if we are at a point where we could start working with the plugins one-by-one.  Are we?

**Comment by leamas** [comment]
> > So if I just do source-state.sh update everything will be taken care of. Is that correct?

> Basically, you need to check in sources.state.

... and to do that sources.state must be updated using  _./source-state.sh save_. I have a test build running after updating sources.state this way. Shall I push it  to main?

**Comment by Rasbats** [comment]
>Shall I push it to main?

Yes.

>Question is if we are at a point where we could start working with the plugins one-by-one. Are we?

How about me pruning site.yml, just leaving the plugins I manage. A recent basic linkcheck showed no errors in ShipDriver manual for instance but the full-linkcheck did. With a pruned sources.state and the trimmed site.yml I can move forward later with the PR for further plugins.

**Comment by leamas** [comment]
OK, sounds good to me.

I just pushed main.

Another thing you perhaps could do is file bugs for say six or nine most important plugins which needs to be fixed. An then assign 2(3) to me, 2(3) to Rick etc.  But then we need @rgleason to be in the loop. 

Thoughts?

**Comment by Rasbats** [comment]
Let me see what happens with my plugins. That would take care of 13/54! Still suspicious of the w3c full linkcheck.
EDIT: I already have forks for the others in my oplaydo account.

**Comment by leamas** [comment]
> Still suspicious of the w3c full linkcheck.

I have worked through the complete development manual, handling more or less all of the reported errors. It has definitely improved the overall quality.

The exception is robots.txt denials and such -- those are handled by extending the list of exceptions in ci/w3c-linkcheck.sh after manual checking.

**Comment by leamas** [comment]
> Let me see what happens with my plugins. That would take care of 13/54! 

Right.

**Comment by rgleason** [comment]
It looks to me like many errors are in sean's plugin docs or ones I handle.

```
weather_routing:ROOT:weather_routing.adoc	Line: 243
statusbar:ROOT:statusbar.adoc 	Line: 252
watchdog:ROOT:watchdog.adoc   	Line: 204
opencpn-master-plugins:squiddio:squiddio.adoc	Line: 171
tactics:ROOT:tactics.adoc     	Line: 216
vdr:ROOT:vdr.adoc             	Line: 159
find-it:ROOT:find-it.adoc     	Line: 156
rotationctrl:ROOT:rotationctrl.adoc	Line: 132
projections:ROOT:projections.adoc	Line: 141
route_great_circle:ROOT:route_great_circle.adoc	Line: 180
celestial_navigation:ROOT:celestial_navigation.adoc	Line: 177
dash-t:ROOT:dash-t.adoc       	Line: 150
rtlsdr:ROOT:rtlsdr.adoc       	Line: 102
pypilot:ROOT:pypilot.adoc     	Line: 264
weatherfax:ROOT:weatherfax.adoc	Line: 234
climatology:ROOT:climatology.adoc	Line: 240
polar:ROOT:polar.adoc         	Line: 222
chartscale:ROOT:chartscale.adoc	Line: 135
ais_radar_display:ROOT:ais_radar_display.adoc	Line: 96
logbook:ROOT:logbook.adoc     	Line: 153
sweep_plot:ROOT:sweep_plot.adoc	Line: 219

```
Perhaps I could edit them individually in their repos?

**Comment by leamas** [comment]
@rgleason Yes, that's the idea. But just to avoid silly problems: Which one(s) do you intend to work with first?

EDIT: we need to make sure which repo to use...

**Comment by rgleason** [comment]
Moved to here

https://github.com/opencpn-manuals/plugins/issues/8#issuecomment-901060228

**Comment by Rasbats** [comment]
@rgleason : 

```
To do: Some of the links to this resource point to broken URI fragments
2021-08-18T15:07:57.3481019Z 	(such as index.html#fragment).
2021-08-18T15:07:57.3481622Z The following fragments need to be fixed:
```
This is probably because they are left in the indexes of available plugins but do not have a source at the moment. When things are looking good in your plugin repo we can add that repo back as a source and we won't get the broken fragment (I hope).

**Comment by Rasbats** [comment]
@leamas : Have disabled edits in site.yml but because it is pulling in other distant plugin sources we still have the "Edit the page" coming up on the built manual. Looking at ways of disabling this. It accounts for a lot of the errors.

**Comment by Rasbats** [comment]
Removed all editing from the pages with the last commit. site.yml changed. It works.

The correction for ShipDriver manual also works - valid links.

**Comment by leamas** [comment]
> Removed all editing from the pages with the last commit. site.yml changed. It works.

This is a great step forward!  A clean starting point is what we need to move on. 

We probably want to enable Edit later, but this is a separate issue, IMHO low priority for now.  Seems that the w3c-linkcheck still reports a lot of errors we need to handle, right?

**Comment by leamas** [comment]
Still, when I build from 4193f8c I get a lot of errors also when running just linkchecker on the result. Is it only me?

**Comment by leamas** [comment]
One problem seems to be about directory structure and attachments. The for example the _JavaScript plugin user guide.pdf_.

It seems to be located in the _assets_ directory:

```
$ find . -name 'JavaScript plugin'\*
./manual/beta-plugins/modules/javascript/assets/JavaScript plugin user guide.pdf
```

However, according to https://docs.antora.org/antora/2.0/component-structure/ it should live in _assets/attachments_ as I read it.

Furthermore, the reference should be using {attachmentsdir} but uses links instead:
```
$ git grep 'JavaScript.*plugin'
/work/plugins/plugins/docs/opencpn-beta-plugins/0.1/javascript/assets/JavaScript plugin user guide.pdf'>
manual/beta-plugins/modules/javascript/pages/javascript.adoc:link:assets/JavaScript%20plugin%20user%20guide.pdf[Javascript plugin user guide]
manual/beta-plugins/modules/javascript/pages/javascript.adoc:https://github.com/antipole2/JavaScript_pi/blob/master/documentation/JavaScript%20plugin%20user%20guide.pdf[Complete

```
So, here is some problems.

**Comment by leamas** [comment]
OTOH, If I drop the manual/beta-plugins from the sources linkchecker runs cleanly. And while w3c-linkchecker still reports a lot of errors for some plugins, many other passes cleanly.

To me, it seems like an idea to drop the beta-plugins for now.

The bulk of my errors comes seems to come from the navbar which is included everywhere.

**Comment by Rasbats** [comment]
>OTOH, If I drop the manual/beta-plugins from the sources linkchecker runs cleanly

Prefer to keep manual/beta-plugins going at the moment. A lot of attachment issues have already been dealt with. A case of using link:../wherever/fileforusing.pdf. WIP.

**Comment by Rasbats** [comment]
The navbar is what might have been causing the edit this page errors. Are we using the same site.yml?

**Comment by leamas** [comment]
The core issue causing so many errors seems to be modules/ROOT/partials/nav-plugins.adoc. It still has references to each and every plugin which have existed causing a lot of link errors. I'm not that fond of commenting out things, but here it might make sense to comment out all links which references yet-not-used plugins.

**Comment by leamas** [comment]
> Are we using the same site.yml?

I'm on 4193f8c, clean index. So, yes.

EDIT: unless you have not-pushed changes, of course.

**Comment by leamas** [comment]
Same problem in modules/ROOT/pages/plugins.adoc, both master and beta.

**Comment by leamas** [comment]
@rasbats: are you about to fix this? If not, I'll  push changes with the offending links commented out...

**Comment by leamas** [comment]
No, I will not push such a big change without talking to you. However, I have pushed to the build branch (forgot full-linkcheck, will add).

Attaching the new output from checklink. Now things looks very sane, many (most?) plugins runs cleanly. With one or two exceptions the remaining errors is just a TODO-list.

[checklink.log.gz](https://github.com/opencpn-manuals/plugins/files/7009990/checklink.log.gz)

**Comment by leamas** [comment]
If I'm not here and you want to copy the build branch to main (out of the top of my head):

    $ git fetch origin build
    $ git show 25bfa1a
    $ git cherry-pick 25bfa1a

**Comment by Rasbats** [comment]
Yes ... that will help. Please push to plugins/main.

**Comment by leamas** [comment]
OK. Done. Build  under way.

**Comment by leamas** [comment]
When I look at the [diagnostics](https://github.com/opencpn-manuals/plugins/runs/3369455463) produced by w3c-linkcheck this looks like a perfectly sane list of things to fix. If you need a hand I can help, it's indeed some work.

Also, if you think something is a false positive I can give you a hand hiding it by amending to the ci script. Without your blessing I'll keep my hands off.

**Comment by leamas** [comment]
@Rasbats ^

**Comment by Rasbats** [comment]
Thanks for the work on the adocs. ShipDriver and Photolayer had edits before your last build and should be fixed. I will work on my others. If you want to pick off any other important plugins like radar_pi that would be good. It would save clashes. 

Out of the house this morning. Continuing PM.

**Comment by leamas** [comment]
I have re-introduced radar_pi without changes, all was OK. Will take a look at aisradar_pi later, please consider it assigned to me.

**Comment by leamas** [comment]
ais_radar in place, no errors. We need to make a PR against Pavel's original repo, but it might depend on #9 before we have the final solution here. There is also #10

**Comment by Rasbats** [comment]
Clash. Already made that PR (accepted). Perhaps change the ref in sources.state unless your updated recently?

**Comment by Rasbats** [comment]
TwoCanPlugin might be good repos next. TwoCanPlugin, EngineDashboard, RacingPlugin.

**Comment by leamas** [comment]
> Clash. Already made that PR (accepted). Perhaps change the ref in sources.state unless your updated recently?

We need to make a new PR. Current changes in plugins/ais_radar_pi  are cleanly on top of nohal's repo, but yet not in place (I added some clean up). Until the new PR is merged we could live with out fork.

**Comment by leamas** [comment]
> TwoCanPlugin might be good repos next. TwoCanPlugin, EngineDashboard, RacingPlugin.

OK, perhaps later... EDIT: I'm on oesenc, which is an VERY important one (using it myself...)

**Comment by Rasbats** [comment]
Raised a new issue "Link fix progress" to record what plugins are TODO and (considered) DONE. I think you can edit the list with your privileges.

**Comment by leamas** [comment]
That was a good way of doing it, I have though of  much worse solutions. I have updated it, it works.

**Comment by Rasbats** [comment]
This is getting painful. I need to check all the dev repos which I have forked to enable PR. Starting a new branch to test those forks.

Sources.state will change to reflect oplaydo rather than rgleason, nohal etc. This will allow me to put stuff back in main that has been tested first.

**Comment by leamas** [comment]
> This is getting painful. I need to check all the dev repos which I have forked to enable PR. 

I don't follow you here...

**Comment by Rasbats** [comment]
The dev repo manuals were forked to enable the manual to be set up. To fix and finally check the links in those manuals as it stands I would need to make a PR, get it accepted and then restore/update/save sources.state. By using my fork as the git in sources.state I can easily update my fork if errors are found, prior to any final PR to the dev.

A lot of the manuals will not have errors and I can move them into main as they stand. This saves polluting main with a lot of plugins with broken links, which was the case earlier.

**Comment by leamas** [comment]
@rasbats: EngineDashboard: Have you the dokuwiki raw translations around somewhere?

**Comment by leamas** [comment]
> The dev repo manuals were forked to....

Ah, right, I'm doing the same (but forking into opencpn-manuals).

**Comment by leamas** [comment]
@Rasbats : OK, found your forks of RacingPlugin and EngineDashboard, they are the starting points. 

It would be great if you could transfer these from your personal accounts to opencpn-manuals so I can update them easily

**Comment by Rasbats** [comment]
[engine-dash.adoc.pdf](https://github.com/opencpn-manuals/plugins/files/7020418/engine-dash.adoc.pdf)

**Comment by Rasbats** [comment]
>It would be great if you could transfer these from your personal accounts to opencpn-manuals so I can update them easily

My turn to be confused. You will need to fork twocan's repos later to make the PR?

**Comment by leamas** [comment]
Not really. Assuming your repo is a fork of twocan I can use that to make the PR (assuming it's ownd by opencpn-manuals).

Let's put this aside, it is not blocking now, I can sort it out later.

EngineDashboard seems to miss some  images engine01.png, engine02.png, engine03.png, engine04.png, engine32.png. You mentioned that you have some images around. Are these among those you have?

**Comment by leamas** [comment]
Sorry for the noise, again. I found them.

**Comment by leamas** [comment]
@Rasbats : Could you please enable your git pre-commit hook? There is trailing whitespace all over the place, making rebasing harder.

**Comment by Rasbats** [comment]
>Could you please enable your git pre-commit hook?

Finally done.

**Comment by leamas** [comment]
This issue is settled. We have a copy of almost all plugins in place, and basically no link errors. Closing
